### DISCUSSION TITLE
유저별 개인 블로그 링크 생성 기능 구현

### DISCUSSION THREAD

**Discussion by rkskekzzz** [80000Coding/80000Coding-Web-Client#47] [Ideas]
# 목표
팔만코딩경의 경우 현재 개인 블로그 처럼 쓸 수 있는 공간이지만, 개인 블로그를 홍보하려고 할 때 다른 작성자의 글도 함께 나와서 불편함이 있습니다. 그래서 복잡하지 않은 방식으로 개인 블로그 링크 생성 기능을 구현하면 어떨까 합니다.

# 방법

### 일반적으로 접속했을 경우
```shell
# 도메인 예시입니다.
https://80000coding.kr/
```
<img width="200" alt="image" src="https://github.com/80000Coding/80000Coding-Web-Client/assets/57925497/56933671-a039-4f48-ad7d-d6aa2b1e22f0">

### 개인 블로그 링크 생성 기능
```
https://80000coding.kr/username
or
https://80000coding.kr/?userId=userid
```
<img width="244" alt="스크린샷 2023-08-05 오후 6 50 12" src="https://github.com/80000Coding/80000Coding-Web-Client/assets/57925497/839c401f-66f8-4d81-8ffe-954ae4a6e13e">

복잡하게 구현하지 않고 기존 디자인을 활용하면 좋을 것 같습니다
백엔드에서도 추가 기능 구현이 필요없어 보입니다.
프론트에서는 라우팅정도만 고려하면 될 것 같습니다.

**Comment by leekh716** [comment]
조씁니다👍 둘 중에 `https://80000coding.kr?userId=userId` 형태로 쿼리로 받는쪽에 한표욧

**Reply by rkskekzzz** [reply]
좋습니다!! 게시글 피드파트 완성되면 작업 해볼게요!

**Comment by ChanHoHan** [comment]
백엔드에서도 queryString을 받는 api endpoint를 추가해야할거 같은데.. 제가 이해한게 맞을까여?

**Reply by rkskekzzz** [reply]
백엔드에서는 게시글 가져올 때 유저 한명의 필터를 걸 수 있으면 될 것 같아요!
```SQL
SELECT * 
FROM Content
WHERE user_id = {특정 사용자의 user_id};
```

**Reply by rkskekzzz** [reply]
어차피 `내가 쓴 글 보기` 기능을 만들려면 필요할 것 같아서 해당 API가 만들어지면 동일하게 사용하면 되지 않을까요?

**Reply by ChanHoHan** [reply]
이게 이번에 처음 제시된 안건이면, api는 두개 있는게 나을거 같아용 ..
그게 아니면, userId null인지 여부로 분기처리 로직이 들어갈거 같은데
다른 백엔드 분들의 의견도 들어볼게요

전체 사용자 게시글 조회
https://80000coding.kr/content

특정 사용자 게시글 조회
https://80000coding.kr/contnet?userId={userId}

**Reply by ChanHoHan** [reply]
아 넵 ~ 동일하게 사용하면 될거같아여
